Mathlib/RingTheory/Polynomial/DegreeLT.lean

```diff
@@ -0,0 +1,218 @@
+    ((addLinearEquiv R m n).symm (LinearMap.inr R _ _ Q) : R[X]) = (Q : R[X]) * X ^ (m : ℕ) := by
+  rw [← (basis ..).sum_repr Q]
+  simp [-LinearMap.coe_inr, Finset.sum_mul, addLinearEquiv_symm_apply_xPow_right,
+    smul_eq_C_mul, mul_assoc, ← pow_add, add_comm]
+
+lemma addLinearEquiv_symm_apply (PQ) :
+    ((addLinearEquiv R m n).symm PQ : R[X]) = (PQ.1 : R[X]) + (PQ.2 : R[X]) * X ^ (m : ℕ) := calc
+  _ = ((addLinearEquiv R m n).symm (LinearMap.inl R _ _ PQ.1 + LinearMap.inr R _ _ PQ.2) : R[X]) :=
+      by rw [LinearMap.inl_apply, LinearMap.inr_apply, Prod.add_def, add_zero, zero_add]
+  _ = _ := by rw [map_add, Submodule.coe_add,
+        addLinearEquiv_symm_apply_left, addLinearEquiv_symm_apply_right]
+
+lemma addLinearEquiv_symm_apply' (PQ) :
+    ((addLinearEquiv R m n).symm PQ : R[X]) = (PQ.1 : R[X]) + X ^ (m : ℕ) * (PQ.2 : R[X]) := by
+  rw [X_pow_mul, addLinearEquiv_symm_apply]
+
+lemma addLinearEquiv_apply' {R : Type*} [Ring R] (f) :
+    ((addLinearEquiv R m n f).1 : R[X]) = f %ₘ (X ^ m) ∧
+      ((addLinearEquiv R m n f).2 : R[X]) = f /ₘ (X ^ m) := by
+  rw [and_comm, eq_comm, eq_comm (b := _ %ₘ _)]
+  nontriviality R; refine div_modByMonic_unique _ _ (monic_X_pow _) ⟨?_, ?_⟩
+  · rw [← addLinearEquiv_symm_apply', LinearEquiv.symm_apply_apply]
+  · rw [degree_X_pow, ← mem_degreeLT]; exact Subtype.prop _
+
+lemma addLinearEquiv_apply_fst {R : Type*} [Ring R] (f) :
+    ((addLinearEquiv R m n f).1 : R[X]) = f %ₘ (X ^ m) :=
+  (addLinearEquiv_apply' f).1
+
+lemma addLinearEquiv_apply_snd {R : Type*} [Ring R] (f) :
+    ((addLinearEquiv R m n f).2 : R[X]) = f /ₘ (X ^ m) :=
+  (addLinearEquiv_apply' f).2
+
+lemma addLinearEquiv_apply {R : Type*} [Ring R] (f) :
+    addLinearEquiv R m n f =
+      (⟨f %ₘ (X ^ m), addLinearEquiv_apply_fst f ▸ Subtype.prop _⟩,
+      ⟨f /ₘ (X ^ m), addLinearEquiv_apply_snd f ▸ Subtype.prop _⟩) :=
+  Prod.ext (Subtype.ext <| addLinearEquiv_apply_fst f) (Subtype.ext <| addLinearEquiv_apply_snd f)
+
+end degreeLT
+
+section taylor
+
+variable {R : Type*} [CommRing R] {r : R} {m n : ℕ} {s : R} {f g : R[X]}
+
+/-- The map `taylor r` induces an automorphism of the module `R[X]_n` of polynomials of
+degree `< n`. -/
+noncomputable def taylorLinear (r : R) (n) : (R[X]_n) ≃ₗ[R] (R[X]_n) where
+  toFun     := fun P ↦ ⟨P.1.taylor r, mem_degreeLT.2 <| (degree_taylor ..).trans_lt <|
+    mem_degreeLT.1 P.2⟩
+  invFun    := fun P ↦ ⟨P.1.taylor (-r), mem_degreeLT.2 <| (degree_taylor ..).trans_lt <|
+    mem_degreeLT.1 P.2⟩
+  map_add'  := fun _ _ ↦ Subtype.ext <| map_add ..
+  map_smul' := fun _ _ ↦ Subtype.ext <| map_smul ..
+  left_inv  := fun _ ↦ Subtype.ext <| (taylorEquiv r).symm_apply_apply _
+  right_inv := fun _ ↦ Subtype.ext <| (taylorEquiv r).apply_symm_apply _
+
+@[simp] lemma taylorLinear_apply (P : R[X]_n) :
+    (taylorLinear r n P : R[X]) = (P : R[X]).taylor r :=
+  rfl
+
+@[simp] theorem taylor_det' :
+    LinearMap.det (taylorLinear r n : (R[X]_n) →ₗ[R] _) = 1 := by
```

Comment:
```suggestion
@[simp] theorem det_taylor_toLinearMap :
    (taylorLinear r n).toLinearMap.det = 1 := by
```
And the lemma below should be `det_taylor` as we call `f a` as `f_a` regardless of dot notation.
(You can argue the same for `toLinearMap` but I think we have an exception for structure projections)